### PR TITLE
feat(deployer): split up deployer error enum

### PR DESCRIPTION
**Pull request by oddgrd**
Split up `deployer/src/error`, since the handlers use almost none of its variants.

### CODE REVIEW

File: deployer/src/handlers/error.rs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+impl Serialize for Error {
+    fn serialize<S>(&self, serializer: S) -> std::result::Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        let mut map = serializer.serialize_map(Some(2))?;
+        map.serialize_entry("type", &format!("{:?}", self))?;
+        map.serialize_entry("msg", &self.source().unwrap().to_string())?;
```
Comment (brokad) [inline deployer/src/handlers/error.rs:23]: As per `thiserror` doc `.source().unwrap()` is safe only for variants that have `#[from]` or `#[source]` so we should be a bit careful - perhaps we can just `.unwrap_or_default` or something?

File: deployer/src/handlers/error.rs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    Persistence(#[from] crate::persistence::PersistenceError),
+}
+
+impl Serialize for Error {
+    fn serialize<S>(&self, serializer: S) -> std::result::Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        let mut map = serializer.serialize_map(Some(2))?;
+        map.serialize_entry("type", &format!("{:?}", self))?;
```
Comment (brokad) [inline deployer/src/handlers/error.rs:22]: I know you moved this over from somewhere else, so not really part of this PR, but I couldn't help but notice:

Why `{:?}` and why `self`? Won't this inline the entire debug disposition of the error as its "type"? Would it be better to have a `match` and display the variant name?

**Review by brokad — APPROVED** [review]
LGTM 😄 

The two comments I have actually came from code you inherited in the diff!

File: deployer/src/handlers/error.rs (hunk 2 above)
Comment (oddgrd) [reply deployer/src/handlers/error.rs:22]: Yeah, this will have the entire debug representation as the "type". With an example error the serialized output will be: `{"message":{"type":"Persistence(Database(PoolTimedOut))","msg":"Database error: pool timed out while waiting for an open connection"}}`. It's a bit verbose and I can change this to just match and return variant name, but I think the explicitness of this is kinda nice for internal errors.

File: deployer/src/handlers/error.rs (hunk 1 above)
Comment (oddgrd) [reply deployer/src/handlers/error.rs:23]: I agree, this is a good point. Since we implement `Display` through `thiserror` for all the deployer errors, we can do `&self.source().unwrap_or(self).to_string()`. This way we will get either the source or the `Display` representation. If we add a new error to `handlers::Error`, this will not compile if the new error doesn't implement `Display`.
